homeassistant/components/osramlightify/light.py

```diff
@@ -4,270 +4,410 @@
         return self._brightness
 
     @property
     def is_on(self):
-        """Update status to True if device is on."""
-        return self._state
+        """Return True if the device is on."""
+        return self._is_on
 
     @property
     def supported_features(self):
-        """Flag supported features."""
-        return SUPPORT_OSRAMLIGHTIFY
+        """List of supported features."""
+        return self._supported_features
 
     @property
     def effect_list(self):
         """List of supported effects."""
-        return [EFFECT_RANDOM]
+        return self._effect_list
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return self._min_mireds
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        return self._max_mireds
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
+
+    @property
+    def state_attributes(self):
+        """Return optional state attributes."""
+        data = {}
+        if self._supported_features & SUPPORT_COLOR_TEMP:
+            data[ATTR_MIN_MIREDS] = self._min_mireds
+            data[ATTR_MAX_MIREDS] = self._max_mireds
+
+        if self._supported_features & SUPPORT_EFFECT:
+            data[ATTR_EFFECT_LIST] = self._effect_list
+
+        if self.is_on:
+            if self._supported_features & SUPPORT_BRIGHTNESS:
+                data[ATTR_BRIGHTNESS] = self._brightness
+
+            if self._supported_features & SUPPORT_COLOR_TEMP:
+                data[ATTR_COLOR_TEMP] = self._color_temp
+
+            if self._supported_features & SUPPORT_COLOR:
+                data[ATTR_RGB_COLOR] = self._rgb_color
+                data[ATTR_HS_COLOR] = self._hs_color
+                data[ATTR_XY_COLOR] = self._xy_color
+
+        return data
+
+    def play_effect(self, effect, transition):
+        """Play selected effect."""
+        if effect == EFFECT_RANDOM:
+            self._rgb_color = (random.randrange(0, 256),
+                               random.randrange(0, 256),
+                               random.randrange(0, 256))
+            self._hs_color = color_util.color_RGB_to_hs(*self._rgb_color)
+            self._xy_color = color_util.color_RGB_to_xy(*self._rgb_color)
+            self._luminary.set_rgb(*self._rgb_color, transition)
+            self._luminary.set_onoff(True)
+            return True
+
+        return False
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
-        if ATTR_TRANSITION in kwargs:
-            transition = int(kwargs[ATTR_TRANSITION] * 10)
-        else:
-            transition = 0
+        transition = int(kwargs.get(ATTR_TRANSITION, 0) * 10)
+        if ATTR_EFFECT in kwargs:
+            if self.play_effect(kwargs[ATTR_EFFECT], transition):
+                self.schedule_update_ha_state()
```

Comment:
You should not need to `schedule_update_ha_state()`, just return.

Reply:
Ok!  I have copied those calls from the original module.

Reply:
All right. Since you said it was a rewrite, I did not look at the old code :)

Reply:
But those calls are not needed and should be removed anyway, correct?

Reply:
Yes, the state is updated by the core after you return.